app/main.py

```diff
@@ -16,4 +16,32 @@
 }
 collection_of_coins = {1, 2, 25}
 
-# write your code here
+sorted_variables = {"mutable": [], "immutable": []}
+
+
+def create_sorted_dictionary(*args) -> dict:
+    mutable, immutable = [], []
+
+    for arg in args:
+        if isinstance(arg, (list, dict, set)):
+            mutable.append(arg)
+        else:
+            immutable.append(arg)
+
+    sorted_variables["mutable"] = mutable
+    sorted_variables["immutable"] = immutable
+    return sorted_variables
```

Comment:
The function `create_sorted_dictionary` modifies the global variable `sorted_variables`. This is not a good practice as it can lead to unexpected behavior if the function is called multiple times. Consider returning a new dictionary from the function instead of modifying the global variable.